### PR TITLE
SDK-436 [C# SDK] refactor HttpDao

### PR DESCRIPTION
**Pull request by VladimirLevitskiy**
Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic (if any):
- [SDK-436](https://incountry.atlassian.net/browse/SDK-436)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

File: UnitTests/storage/StorageAttachmentTest.cs [hunk 1]
```diff
@@ -222,7 +226,7 @@ public void GetAttachmentNegativeTest()
             var serverException = Assert.ThrowsAsync<StorageServerException>(() =>
                 storage.GetAttachmentFileAsync(Country, RecordKey, FileId));
             Assert.IsTrue(serverException.Message.Contains(
-                "Unexpected response: StatusCode: 405, ReasonPhrase: 'Method Not Allowed'",
+                "Unexpected response: code=405, url=http://localhost:8087/v2/storage/records/us/3e758b0dc7d6a52f2a314fa996bf5513be8bc5d276935ce2628c67151abc6da3/attachments/456, content=Error",
```
Comment (RommeroInCountry) [inline UnitTests/storage/StorageAttachmentTest.cs:229]: We won't miss useful error information with this PR, will we? Like "Method Not Allowed" or any other message returned by the server

File: InCountrySDK/Implementation/Http/HttpDao.cs [hunk 2]
```diff
@@ -157,52 +158,110 @@ public class HttpDao : ThrottlingRequestExecutor, IDao
...
+            using (var request = new HttpRequestMessage(httpMethod, new Uri(url)))
+            {
+                request.Content = CreateContent(content, stringBody, contentType, dataStream, fileName);
+                if (IsNullOrEmpty(token))
+                {
+                    token = await _tokenClient.RefreshTokenAsync(false, pop.Audience, pop.Region).ConfigureAwait(false);
+                }
+
+                request.Headers.Authorization = new AuthenticationHeaderValue(Bearer, token);
+                request.Headers.TryAddWithoutValidation(XEnvId, _environmentId);
```
Comment (RommeroInCountry) [inline InCountrySDK/Implementation/Http/HttpDao.cs:176]: Please consider renaming to XEnvId -> XEnvIdHeader, Bearer -> AuthHeaderTypePrefix

File: InCountrySDK/Implementation/Http/HttpDao.cs [hunk 3]
```diff
@@ -405,36 +440,31 @@ private string GetAudienceForMiniPop(string mainUrl, string country)
...
+                throw new StorageServerException(message);
             }
             catch (StorageException)
             {
                 throw;
             }
+
             catch (System.Exception ex)
             {
                 throw CreateException(ex, address);
```
Comment (RommeroInCountry) [inline InCountrySDK/Implementation/Http/HttpDao.cs:470]: Can we somehow get rid of such duplication for each method?

**Review by RommeroInCountry — APPROVED** [review]

File: InCountrySDK/Implementation/Http/HttpDao.cs (hunk 2 above)
Comment (VladimirLevitskiy) [reply InCountrySDK/Implementation/Http/HttpDao.cs:176]: will be done at https://github.com/incountry/sdk-csharp/pull/111

File: UnitTests/storage/StorageAttachmentTest.cs (hunk 1 above)
Comment (VladimirLevitskiy) [reply UnitTests/storage/StorageAttachmentTest.cs:229]: No. It was just the name of a response status

File: InCountrySDK/Implementation/Http/HttpDao.cs (hunk 3 above)
Comment (VladimirLevitskiy) [reply InCountrySDK/Implementation/Http/HttpDao.cs:470]: This code will be refactored while the exception handling refactoring task

**Review by robert-par — APPROVED** [review]

**Pull request merged**